auto_round/auto_quantizer.py

```diff
@@ -316,8 +317,11 @@ def convert_model(self, model: nn.Module):
         return model
 
     def _dynamic_import_inference_linear(self, bits, backend):
-        if bits == 4 and self.exllama2_available and "exllamav2" in backend:
-            from auto_round_extension.cuda.qliner_exllamav2 import QuantLinear
+        if bits == 4 and "exllamav2" in backend:
+            if self.exllama2_available:
+                from auto_round_extension.cuda.qliner_exllamav2 import QuantLinear
+            else:
+                return qlinear_qbits.QuantLinear
```

Comment:
as qbits is only for cpu, could we check the model device to determine which backend is used?

Reply:
in [9cbc870](https://github.com/intel/auto-round/pull/145/commits/9cbc870c4f380419101dab923630383ee5ca4eb9) we will import default exllamav2 qlinear and then check whether the qweight tensor on cpu device, if true we will switch the qlinear to qbis_linear.